src/D2L.Bmx/BmxConfigProvider.cs

```diff
@@ -55,12 +57,25 @@ internal class BmxConfigProvider : IBmxConfigProvider {
 	}
 
 	public void SaveConfiguration( BmxConfig config ) {
-
 		if( !Directory.Exists( BmxPaths.BMX_DIR ) ) {
 			Directory.CreateDirectory( BmxPaths.BMX_DIR );
 		}
+		//File permission is only applied on Create, so must delete old file
+		try {
+			if( File.Exists( BmxPaths.CONFIG_FILE_NAME ) ) {
+				File.Delete( BmxPaths.CONFIG_FILE_NAME );
+			}
+		} catch( IOException IOException ) {
+			//maybe we cannot delete it somehow
+		}
 
-		using var writer = new StreamWriter( BmxPaths.CONFIG_FILE_NAME, append: false );
+		var op = new FileStreamOptions();
+		op.Mode = FileMode.Create; //Append false, overrite old
+		op.Access = FileAccess.ReadWrite;
+		if( !RuntimeInformation.IsOSPlatform( OSPlatform.Windows ) ) {
+			op.UnixCreateMode = UnixFileMode.SetUser | UnixFileMode.UserRead | UnixFileMode.UserWrite;
+		}
```

Comment:
We don't need `SetUser` right? That's SUID https://www.redhat.com/sysadmin/suid-sgid-sticky-bit

I'm also thinking of using [conditional compilation](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives#conditional-compilation) for this, i.e.
```csharp
		var op = new FileStreamOptions {
			Mode = FileMode.Create,
			Access = FileAccess.Write,
#if !WINDOWS
			UnixCreateMode = UnixFileMode.UserRead | UnixFileMode.UserWrite,
#endif
		};
```
This moves OS detection from runtime to compile time, which is kinda nice. I don't feel strongly though.